blockcache/blockcache_test.go

```diff
@@ -27,12 +27,24 @@ func (m *mockChainBackend) addBlock(block *wire.MsgBlock, nonce uint32) {
 	hash := block.Header.BlockHash()
 	m.blocks[hash] = block
 }
+
 func (m *mockChainBackend) GetBlock(blockHash *chainhash.Hash) (*wire.MsgBlock, error) {
+	m.incrementCallCount()
+	return m.getBlock(blockHash)
+}
+
+func (m *mockChainBackend) incrementCallCount() {
+	m.Lock()
+	defer m.Unlock()
+	m.chainCallCount++
+}
```

Comment:
all we need to do here is change the GetBlock locks to write locks. The separation of the functions is not needed and in fact it makes the GetBlock function non-atomic.